src/ShaderAsset.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <string>
+
+#include "GL/glew.h"
+#include "SDL_opengl.h"
+
+class ShaderAsset {
+public:
+    bool load_shader(const std::string& vertex_path, const std::string& fragment_path);
+
+    bool valid_;
+    GLuint program_id_;
```

Comment:
By making these public, other classes have freedom to change these as they will. They would be beteer as private with a getter.